interpreter/interpreter.go

```diff
@@ -4388,73 +4388,86 @@ func checkValue(
 	return
 }
 
-func authAccountSaveFunction(
+func authAccountStorageSaveFunction(
 	context FunctionCreationContext,
 	storageValue *SimpleCompositeValue,
 	addressValue AddressValue,
 ) BoundFunctionValue {
 
-	// Converted addresses can be cached and don't have to be recomputed on each function invocation
-	address := addressValue.ToAddress()
-
 	return NewBoundHostFunctionValue(
 		context,
 		storageValue,
 		sema.Account_StorageTypeSaveFunctionType,
 		func(_ *SimpleCompositeValue, invocation Invocation) Value {
 			interpreter := invocation.InvocationContext
+			arguments := invocation.Arguments
+			locationRange := invocation.LocationRange
 
-			value := invocation.Arguments[0]
+			return StorageSave(
+				interpreter,
+				arguments,
+				addressValue,
+				locationRange,
+			)
+		},
+	)
+}
 
-			path, ok := invocation.Arguments[1].(PathValue)
-			if !ok {
-				panic(errors.NewUnreachableError())
-			}
+func StorageSave(
+	context InvocationContext,
+	arguments []Value,
```

Comment:
Here and for similar functions: Maybe the arguments can be extracted at the call-site / out of the invocation arguments slice, and passed as "proper"/typed arguments. Basically, can we maybe replace this parameter with the two, `value Value` and `path PathValue`?

Reply:
Yeah, exactly. Followed this pattern in all stdlib functions so that the arguments extraction and validation (e.g: type casting) doesn't needed to be repeated. I'll capture this in a separate issue as a tech-debt, and maybe leave this as-is for now. Can clean up everything together at once.